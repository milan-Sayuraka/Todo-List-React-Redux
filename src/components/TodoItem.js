import React from 'react';
import { useNavigate } from "react-router-dom";
import { useSelector, useDispatch } from 'react-redux';

//Single todo item component
const TodoItem = (props) => {
    const navigate = useNavigate();
    //Get todoList from todoReducer
    const todoList = useSelector(state => state.todos.todoList)
    //Use for all the dispatch actions
    const dispatch = useDispatch();

    //Remove single todo in the list
    const removeTodoItem = (todoId) => {
        console.log('remove call', todoId)
        //filter to get the todoId which need to be remove
        let newTodoList = todoList.filter(item => item.id !== todoId);
        // dispatch
        dispatch({ type: 'REMOVE_TODO', payload: newTodoList })

    }

    return (
        <li className="collection-item" key={props.item.id}>{props.item.content}

            <span
                onClick={() => {
                    // remove todo items
                    removeTodoItem(props.item.id)
                }}
                className="secondary-content">
                <i className="remove-btn material-icons blue-text">clear</i>
            </span>
            <span
                onClick={() => {
                    // navigate to update screen
                    navigate('/update', { state: props.item });
                }}
                className="secondary-content">
                <i className="material-icons blue-text">edit</i>
            </span>
        </li>
    );
}

export default TodoItem;