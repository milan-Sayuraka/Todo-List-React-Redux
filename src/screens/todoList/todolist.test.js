import renderer from 'react-test-renderer';
import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import TodoItem from '../../components/TodoItem';
import Search from '../../components/search/search'
import TodoList from './todolist';

jest.mock('react-redux');
jest.mock('../../components/TodoItem');
jest.mock('../../components/search/search');

const renderTree = tree => renderer.create(tree);
describe('<TodoList>', () => {
  it('should render component', () => {
    expect(renderTree(<TodoList 
    />).toJSON()).toMatchSnapshot();
  });
  
});