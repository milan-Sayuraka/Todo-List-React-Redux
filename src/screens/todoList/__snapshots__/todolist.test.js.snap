// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoList> should render component 1`] = `
<section
  id="section-todo"
>
  <h3
    className="center-align white-text blue"
  >
    Todo List
  </h3>
  <p
    className="center-align"
  >
    You don't have anything to do!
  </p>
</section>
`;
